optuna/terminator/serror.py

```diff
@@ -0,0 +1,74 @@
+import abc
+from typing import List
+
+import numpy as np
+
+from optuna._experimental import experimental_class
+from optuna.study.study import Study
+from optuna.trial import Trial
+from optuna.trial._state import TrialState
+
+
+_CROSS_VALIDATION_SCORES_KEY = "terminator:cv_scores"
+
+
+class BaseStatisticalErrorEvaluator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def evaluate(
+        self,
+        study: Study,
+    ) -> float:
+        pass
+
+
+@experimental_class("3.2.0")
+class CrossValidationStatisticalErrorEvaluator(BaseStatisticalErrorEvaluator):
+    def evaluate(
+        self,
+        study: Study,
+    ) -> float:
+        if len(study.get_trials(states=(TrialState.COMPLETE,))) == 0:
+            raise ValueError(
+                "Statistical error cannot be calculated because no trial has been completed."
+            )
```

Comment:
Do you think it is enough to check that the `min_n_trials` argument of `TerminatorCallback` is greater than 0? Not a strong opinion, but it can reduce a storage API call.

Reply:
Thanks. Let me keep the assertion because this is a public API and the user technically calls it with an empty study.
Meanwhile, I removed the `ValueError` and just put `assert`. https://github.com/optuna/optuna/pull/4405/commits/e93a62571272d4dfc32e3435aa0742d4e87abd61